internal/edge_cluster/edge_node_subresource.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 Broadcom. All Rights Reserved.
+// SPDX-License-Identifier: MPL-2.0
+
+package edge_cluster
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func EdgeNodeSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the edge node",
+			},
+			"cluster_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The id of the compute cluster",
+			},
+			"admin_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The administrator password for the edge node",
+			},
+			"audit_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The audit password for the edge node",
+			},
+			"root_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The root user password for the edge node",
+			},
+			"tep1_ip": {
```

Comment:
Thanks, good suggestion.

I am going to add a custom validation function though because we are only interested in IPv4 addresses and net.IsCIDR accepts both v6 and v4